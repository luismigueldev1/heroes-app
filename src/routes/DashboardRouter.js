import React from "react";
import { Switch, Route, Redirect } from "react-router-dom";

import NavBar from "../components/NavBar";

import HeroPage from "../pages/HeroPage";
import DcPage from "../pages/DcPage";
import MarvelPage from "../pages/MarvelPage";
import SearchPage from "../pages/SearchPage";

export default function DashboardRouter() {
  return (
    <>
      <NavBar />
      <div className="container mt-3">
        <Switch>
          <Route exact path="/dc/" component={DcPage} />
          <Route exact path="/marvel/" component={MarvelPage} />
          <Route exact path="/hero/:heroid" component={HeroPage} />
          <Route exact path="/search" component={SearchPage} />
          <Redirect to="/dc" />
        </Switch>
      </div>
    </>
  );
}
