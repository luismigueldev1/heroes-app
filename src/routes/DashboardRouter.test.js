import { mount } from "enzyme";
import { MemoryRouter } from "react-router";
import { AuthContext } from "../auth/AuthContext";
import DashboardRouter from "./DashboardRouter";

describe("Pruebas en <DashboardRouter/>", () => {
  const contextValues = {
    dispatch: jest.fn(),
    user: {
      logged: true,
      name: "Luis Miguel",
    },
  };

  test("debe mostrarse correctamente ", () => {
    const wrapper = mount(
      <AuthContext.Provider value={contextValues}>
        <MemoryRouter>
          <DashboardRouter />
        </MemoryRouter>
      </AuthContext.Provider>
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(".text-info").text()).toBe(contextValues.user.name);
  });
});
