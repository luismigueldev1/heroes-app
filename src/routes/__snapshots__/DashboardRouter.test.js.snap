// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <DashboardRouter/> debe mostrarse correctamente  1`] = `
Array [
  <nav
    className="navbar navbar-expand-lg navbar-dark bg-dark"
  >
    <div
      className="container-fluid"
    >
      <a
        className="navbar-brand"
        href="/"
        onClick={[Function]}
      >
        HeroesApp
      </a>
      <button
        aria-controls="navbarNav"
        aria-expanded="false"
        aria-label="Toggle navigation"
        className="navbar-toggler"
        data-bs-target="#navbarNav"
        data-bs-toggle="collapse"
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        className="collapse navbar-collapse"
        id="navbarNav"
      >
        <ul
          className="navbar-nav me-auto"
        >
          <li
            className="nav-item "
          >
            <a
              aria-current="page"
              className="nav-link active"
              href="/dc"
              onClick={[Function]}
              style={Object {}}
            >
              DC
            </a>
          </li>
          <li
            className="nav-item "
          >
            <a
              aria-current={null}
              className="nav-link"
              href="/marvel"
              onClick={[Function]}
            >
              Marvel
            </a>
          </li>
          <li
            className="nav-item "
          >
            <a
              aria-current={null}
              className="nav-link"
              href="/search"
              onClick={[Function]}
            >
              Search
            </a>
          </li>
        </ul>
        <ul
          className="navbar-nav"
        >
          <li
            className="nav-item nav-link"
          >
            <span
              className="text-info"
            >
              Luis Miguel
            </span>
          </li>
          <li
            className="nav-item"
          >
            <button
              className="btn btn-secondary"
              onClick={[Function]}
            >
              Logout
            </button>
          </li>
        </ul>
      </div>
    </div>
  </nav>,
  <div
    className="container mt-3"
  >
    <div>
      <h1>
        DC Heroes
      </h1>
      <hr />
      <div
        className="row animate__animated animate__fadeIn"
      >
        <div
          className=" col-sm-6 col-md-4  col-lg-3 mt-2"
        >
          <div
            className="card"
          >
            <img
              alt="Batman"
              className="card-img"
              src="./assets/heroes/dc-batman.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Batman
              </h5>
              <p
                className="card-text"
              >
                Bruce Wayne
              </p>
              <p
                className="card-text "
              >
                <small
                  className="text-muted"
                >
                   
                  Detective Comics #27
                </small>
              </p>
              <a
                className="btn btn-primary"
                href="/hero/dc-batman"
                onClick={[Function]}
              >
                More...
              </a>
            </div>
          </div>
        </div>
        <div
          className=" col-sm-6 col-md-4  col-lg-3 mt-2"
        >
          <div
            className="card"
          >
            <img
              alt="Superman"
              className="card-img"
              src="./assets/heroes/dc-superman.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Superman
              </h5>
              <p
                className="card-text"
              >
                Kal-El
              </p>
              <p
                className="card-text "
              >
                <small
                  className="text-muted"
                >
                   
                  Action Comics #1
                </small>
              </p>
              <a
                className="btn btn-primary"
                href="/hero/dc-superman"
                onClick={[Function]}
              >
                More...
              </a>
            </div>
          </div>
        </div>
        <div
          className=" col-sm-6 col-md-4  col-lg-3 mt-2"
        >
          <div
            className="card"
          >
            <img
              alt="Flash"
              className="card-img"
              src="./assets/heroes/dc-flash.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Flash
              </h5>
              <p
                className="card-text"
              >
                Jay Garrick
              </p>
              <p
                className="card-text "
              >
                <small
                  className="text-muted"
                >
                   
                  Flash Comics #1
                </small>
              </p>
              <a
                className="btn btn-primary"
                href="/hero/dc-flash"
                onClick={[Function]}
              >
                More...
              </a>
            </div>
          </div>
        </div>
        <div
          className=" col-sm-6 col-md-4  col-lg-3 mt-2"
        >
          <div
            className="card"
          >
            <img
              alt="Green Lantern"
              className="card-img"
              src="./assets/heroes/dc-green.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Green Lantern
              </h5>
              <p
                className="card-text"
              >
                Alan Scott
              </p>
              <p
                className="card-text "
              >
                <small
                  className="text-muted"
                >
                   
                  All-American Comics #16
                </small>
              </p>
              <a
                className="btn btn-primary"
                href="/hero/dc-green"
                onClick={[Function]}
              >
                More...
              </a>
            </div>
          </div>
        </div>
        <div
          className=" col-sm-6 col-md-4  col-lg-3 mt-2"
        >
          <div
            className="card"
          >
            <img
              alt="Green Arrow"
              className="card-img"
              src="./assets/heroes/dc-arrow.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Green Arrow
              </h5>
              <p
                className="card-text"
              >
                Oliver Queen
              </p>
              <p
                className="card-text "
              >
                <small
                  className="text-muted"
                >
                   
                  More Fun Comics #73
                </small>
              </p>
              <a
                className="btn btn-primary"
                href="/hero/dc-arrow"
                onClick={[Function]}
              >
                More...
              </a>
            </div>
          </div>
        </div>
        <div
          className=" col-sm-6 col-md-4  col-lg-3 mt-2"
        >
          <div
            className="card"
          >
            <img
              alt="Wonder Woman"
              className="card-img"
              src="./assets/heroes/dc-wonder.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Wonder Woman
              </h5>
              <p
                className="card-text"
              >
                Princess Diana
              </p>
              <p
                className="card-text "
              >
                <small
                  className="text-muted"
                >
                   
                  All Star Comics #8
                </small>
              </p>
              <a
                className="btn btn-primary"
                href="/hero/dc-wonder"
                onClick={[Function]}
              >
                More...
              </a>
            </div>
          </div>
        </div>
        <div
          className=" col-sm-6 col-md-4  col-lg-3 mt-2"
        >
          <div
            className="card"
          >
            <img
              alt="Martian Manhunter"
              className="card-img"
              src="./assets/heroes/dc-martian.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Martian Manhunter
              </h5>
              <p
                className="card-text"
              >
                J'onn J'onzz
              </p>
              <p
                className="card-text "
              >
                <small
                  className="text-muted"
                >
                   
                  Detective Comics #225
                </small>
              </p>
              <a
                className="btn btn-primary"
                href="/hero/dc-martian"
                onClick={[Function]}
              >
                More...
              </a>
            </div>
          </div>
        </div>
        <div
          className=" col-sm-6 col-md-4  col-lg-3 mt-2"
        >
          <div
            className="card"
          >
            <img
              alt="Robin/Nightwing"
              className="card-img"
              src="./assets/heroes/dc-robin.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Robin/Nightwing
              </h5>
              <p
                className="card-text"
              >
                Dick Grayson
              </p>
              <p
                className="card-text "
              >
                <small
                  className="text-muted"
                >
                   
                  Detective Comics #38
                </small>
              </p>
              <a
                className="btn btn-primary"
                href="/hero/dc-robin"
                onClick={[Function]}
              >
                More...
              </a>
            </div>
          </div>
        </div>
        <div
          className=" col-sm-6 col-md-4  col-lg-3 mt-2"
        >
          <div
            className="card"
          >
            <img
              alt="Blue Beetle"
              className="card-img"
              src="./assets/heroes/dc-blue.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Blue Beetle
              </h5>
              <p
                className="card-text"
              >
                Dan Garret
              </p>
              <p
                className="card-text "
              >
                <small
                  className="text-muted"
                >
                   
                  Mystery Men Comics #1
                </small>
              </p>
              <a
                className="btn btn-primary"
                href="/hero/dc-blue"
                onClick={[Function]}
              >
                More...
              </a>
            </div>
          </div>
        </div>
        <div
          className=" col-sm-6 col-md-4  col-lg-3 mt-2"
        >
          <div
            className="card"
          >
            <img
              alt="Black Canary"
              className="card-img"
              src="./assets/heroes/dc-black.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Black Canary
              </h5>
              <p
                className="card-text"
              >
                Dinah Drake
              </p>
              <p
                className="card-text "
              >
                <small
                  className="text-muted"
                >
                   
                  Flash Comics #86
                </small>
              </p>
              <a
                className="btn btn-primary"
                href="/hero/dc-black"
                onClick={[Function]}
              >
                More...
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
